### PR TITLE
Improve git commands and version handling

### PR DESCRIPTION
**Pull request by Jafaral**
Move git commands to a script for better handling of out of tree
builds. Refactor/cleanup some rules. Update the connfigure script
to use the new version script. Writeo ut two new summary files:
 * Configuration summary: unicon-config.log
 * Features Summary: unicon-features.log

Signed-off-by: Jafar Al-Gharaibeh <to.jafar@gmail.com>

### CODE REVIEW

**Review by Don-Ward — APPROVED** [review]
TL;DR It's all working as expected and can also handle a release candidate build if we decide to do that in the future. 

My only (minor) criticism is that we have two places to keep the version number in sync: one in src/h/version.h and the other in config/scripts/version.sh

I tested this on my local machine and got the following version strings by changing the git meta-data (and rebuilding). Users  who take the release build will only ever see the first two strings. Users who build from source will also see the third string. Only developers are likely to see the rest.

On master with annotated tag 13.2
  Unicon Version 13.2.  October 15, 2020 (Release)
On branch 13.2 with annotated tag 13.2.2
  Unicon Version 13.2.  October 15, 2020 (13.2.2 Maintenance Release)
On master (untagged)
  Unicon Version 13.2.  October 15, 2020 (13.1-341-g8ca5d681 pre-release)
On master with lightweight tag 13.2-RC1
  Unicon Version 13.2.  October 15, 2020 (13.1-341-g8ca5d681 pre-release)
On master with annotated tag 13.2-RC1
  Unicon Version 13.2.  October 15, 2020 (13.2-RC1-0-g8ca5d681 pre-release)
On branch 13.2 with annotated tag 13.2.2-RC1
  Unicon Version 13.2.  October 15, 2020 (13.2.2-RC1-0-g8ca5d681 Maintenance pre-release)
On branch PR150 (untagged)
  Unicon Version 13.2.  October 15, 2020 (13.1-341-g8ca5d681 Development "PR150")
On branch PR150 with annotated tag 13.2
  Unicon Version 13.2.  October 15, 2020 (13.2-0-g8ca5d681 Development "PR150")
